### PR TITLE
Fixes the consistency for the capitalization of 'Agent' -> 'agent' across the project

### PR DESCRIPTION
**Pull request by dinsley**
This fixes the consistent usage of the word 'Agent' to 'agent' across the project that is outlined in the issue opened here: 

https://github.com/newrelic/newrelic-ruby-agent/issues/1118

# Reviewer Checklist
- [ ] Perform code review
- [ ] Add performance label
- [ ] Perform appropriate level of performance testing
- [ ] Confirm all checks passed
- [ ] Add version label prior to acceptance

### CODE REVIEW

File: lib/new_relic/agent/worker_loop.rb [hunk 1]
```diff
@@ -88,7 +88,7 @@ def run_task
             raise
           rescue => e
             # Don't blow out the stack for anything that hasn't already propagated
-            ::NewRelic::Agent.logger.error('Error running task in Agent Worker Loop:', e)
+            ::NewRelic::Agent.logger.error('Error running task in agent Worker Loop:', e)
```
Comment (kaylareopelle) [inline lib/new_relic/agent/worker_loop.rb:91]:
```suggestion
            ::NewRelic::Agent.logger.error('Error running task in agent worker loop:', e)
```

File: lib/new_relic/agent/event_loop.rb [hunk 2]
```diff
@@ -156,7 +156,7 @@ def dispatch_event(event, args)
         end
 
         if !errors.empty?
-          ::NewRelic::Agent.logger.error("#{errors.size} error(s) running task for event '#{event}' in Agent Event Loop:", *errors)
+          ::NewRelic::Agent.logger.error("#{errors.size} error(s) running task for event '#{event}' in agent Event Loop:", *errors)
```
Comment (kaylareopelle) [inline lib/new_relic/agent/event_loop.rb:159]:
```suggestion
          ::NewRelic::Agent.logger.error("#{errors.size} error(s) running task for event '#{event}' in agent event loop:", *errors)
```

File: init.rb [hunk 3]
```diff
@@ -12,7 +12,7 @@
 # If you can't find any log files and you don't see anything in your
 # application log files please visit support.newrelic.com.
 
-# Initializer for the NewRelic Ruby Agent
+# Initializer for the New Relic Ruby agent
```
Comment (kaylareopelle) [inline init.rb:15]: Good catch!

File: newrelic.yml [hunk 4]
```diff
@@ -8,8 +8,8 @@
 # For full documentation of agent configuration options, please refer to
 # https://docs.newrelic.com/docs/agents/ruby-agent/installation-configuration/ruby-agent-configuration
 
-common: &default_settings
-  # Required license key associated with your New Relic account.
+common:
+  &default_settings # Required license key associated with your New Relic account.
```
Comment (kaylareopelle) [inline newrelic.yml:12]:
```suggestion
common: &default_settings 
# Required license key associated with your New Relic account.
```

I think `common:` and  `&default_settings` need to be on the same line so the node mapping works. Did this indentation strategy work for you?

File: newrelic.yml [hunk 5]
```diff
@@ -341,7 +341,7 @@ common: &default_settings
   # embedded service within another framework and the agent is detecting the Sinatra
   # app and skipping the at_exit handler as a result. Sinatra classically runs the
   # entire application in an at_exit block and would otherwise misbehave if the
-  # Agent's at_exit handler was also installed in those circumstances. Note:
+  # agent's at_exit handler was also installed in those circumstances. Note:
```
Comment (kaylareopelle) [inline newrelic.yml:344]: The `newrelic.yml` file is autogenerated by the `newrelic:update_newrelicyml` rake task. The descriptions for the config values originate in [`lib/new_relic/agent/configuration/default_source.rb`](https://github.com/newrelic/newrelic-ruby-agent/blob/dev/lib/new_relic/agent/configuration/default_source.rb). I don't believe that file was updated as part of this PR. Could you update the `force_install_exit_handler` description in that file?

File: lib/new_relic/agent/attribute_filter.rb [hunk 6]
```diff
@@ -2,7 +2,7 @@
 # See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
 # frozen_string_literal: true
 
-# This class applies filtering rules as specified in the Agent Attributes
+# This class applies filtering rules as specified in the agent Attributes
```
Comment (kaylareopelle) [inline lib/new_relic/agent/attribute_filter.rb:5]: This is one of a few places where I found some additional capitalization following `agent` that I don't think we need anymore.

```suggestion
# This class applies filtering rules as specified in the agent attributes
```

File: lib/new_relic/agent/attribute_filter.rb [hunk 7]
```diff
@@ -44,7 +44,7 @@
 # 1. First, the names are compared lexicographically. This has the impact of
 #    forcing shorter (more general) rules towards the top of the list and longer
 #    (more specific) rules towards the bottom. This is important, because the
-#    Agent Attributes spec stipulates that the most specific rule for a given
+#    agent Attributes spec stipulates that the most specific rule for a given
```
Comment (kaylareopelle) [inline lib/new_relic/agent/attribute_filter.rb:47]:
```suggestion
#    agent attributes spec stipulates that the most specific rule for a given
```

**Review by kaylareopelle — COMMENTED** [review]
This looks really good! Thank you! I think we have a few small things to change before it's ready to go.

**Comment by dinsley** [comment]
Will go through and fix this one up this evening!

File: newrelic.yml (hunk 4 above)
Comment (dinsley) [reply newrelic.yml]: this ended being a formatter that was operating on save -- it looks like it's valid, but it was unintentional, so moved it back to how it was anyways.

File: newrelic.yml (hunk 5 above)
Comment (dinsley) [reply newrelic.yml:343]: updated this one -- thanks for pointing these ones out that I missed!

**Comment by dinsley** [comment]
should be g2g for another review pass now :)

**Review by kaylareopelle — APPROVED** [review]
Looks great! Thank you for your help, @dinsley! 🚀

**Pull request merged**